vsphere/resource_vsphere_compute_cluster_host_group.go

```diff
@@ -0,0 +1,375 @@
+	if err != nil {
+		return err
+	}
+
+	info, err := resourceVSphereComputeClusterHostGroupFindEntry(cluster, name)
+	if err != nil {
+		return err
+	}
+
+	if info == nil {
+		// The configuration is missing, blank out the ID so it can be re-created.
+		d.SetId("")
+		return nil
+	}
+
+	// Save the compute_cluster_id and name here. These are
+	// ForceNew, but we set these for completeness on import so that if the wrong
+	// cluster/VM combo was used, it will be noted.
+	if err = d.Set("compute_cluster_id", cluster.Reference().Value); err != nil {
+		return fmt.Errorf("error setting attribute \"compute_cluster_id\": %s", err)
+	}
+
+	// This is the "correct" way to set name here, even if it's a bit
+	// superfluous.
+	if err = d.Set("name", info.Name); err != nil {
+		return fmt.Errorf("error setting attribute \"name\": %s", err)
+	}
+
+	if err = flattenClusterHostGroup(d, info); err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: Read completed successfully", resourceVSphereComputeClusterHostGroupIDString(d))
+	return nil
+}
+
+func resourceVSphereComputeClusterHostGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning update", resourceVSphereComputeClusterHostGroupIDString(d))
+
+	cluster, name, err := resourceVSphereComputeClusterHostGroupObjects(d, meta)
+	if err != nil {
+		return err
+	}
+
+	info, err := expandClusterHostGroup(d, name)
+	if err != nil {
+		return err
+	}
+	spec := &types.ClusterConfigSpecEx{
+		GroupSpec: []types.ClusterGroupSpec{
+			{
+				ArrayUpdateSpec: types.ArrayUpdateSpec{
+					Operation: types.ArrayUpdateOperationEdit,
+				},
+				Info: info,
+			},
+		},
+	}
+
+	if err := clustercomputeresource.Reconfigure(cluster, spec); err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: Update finished successfully", resourceVSphereComputeClusterHostGroupIDString(d))
+	return resourceVSphereComputeClusterHostGroupRead(d, meta)
+}
+
+func resourceVSphereComputeClusterHostGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	log.Printf("[DEBUG] %s: Beginning delete", resourceVSphereComputeClusterHostGroupIDString(d))
+
+	cluster, name, err := resourceVSphereComputeClusterHostGroupObjects(d, meta)
+	if err != nil {
+		return err
+	}
+
+	spec := &types.ClusterConfigSpecEx{
+		GroupSpec: []types.ClusterGroupSpec{
+			{
+				ArrayUpdateSpec: types.ArrayUpdateSpec{
+					Operation: types.ArrayUpdateOperationRemove,
+					RemoveKey: name,
+				},
+			},
+		},
+	}
+
+	if err := clustercomputeresource.Reconfigure(cluster, spec); err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] %s: Deleted successfully", resourceVSphereComputeClusterHostGroupIDString(d))
+	return nil
+}
+
+func resourceVSphereComputeClusterHostGroupImport(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	var data map[string]string
+	if err := json.Unmarshal([]byte(d.Id()), &data); err != nil {
+		return nil, err
```

Comment:
add some additional context on this `err` before returning?

Reply:
This will just return the JSON error which will hopefully be clear enough to the user. I think there are a couple of other resources that have similar import patterns that don't necessarily include any additional error context, so might as well just leave it for consistency for now.